roaster-selenide/roaster-selenide-api/src/main/java/hu/icellmobilsoft/roaster/selenide/SelenideConfigExtension.java

```diff
@@ -0,0 +1,91 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.roaster.selenide;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.Extension;
+
+import com.codeborne.selenide.Configuration;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.WebDriverRunner;
+import hu.icellmobilsoft.coffee.se.logging.Logger;
+import hu.icellmobilsoft.roaster.selenide.drivers.ChromeDeviceModeDriverProvider;
+import hu.icellmobilsoft.roaster.selenide.drivers.CustomIEDriverProvider;
+import hu.icellmobilsoft.roaster.selenide.enums.Browser;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.weld.environment.se.events.ContainerInitialized;
+
+/**
+ * Selenide Configuration extension.
+ *
+ * @author speter555
+ * @since 0.2.0
+ */
+public class SelenideConfigExtension implements Extension {
+
+    private Logger logger = Logger.getLogger(SelenideConfigExtension.class);
+
+    /**
+     * Handle container Initialized event
+     *
+     * @param containerInitialized instance
+     */
+    public void observesContainerInitialized(@Observes @Initialized(ApplicationScoped.class) ContainerInitialized containerInitialized) {
+        initDriver();
+    }
+
+    private void initDriver() {
+        logger.debug(">> initDriver()");
+        Config config = ConfigProvider.getConfig();
+        Boolean deviceMode = config.getOptionalValue("browser.device.mode", Boolean.class).orElse(false);
+        String browserType = config.getOptionalValue("browser.type", String.class).orElse(Browser.CHROME.name());
+        boolean isRemote = BooleanUtils.isTrue(config.getOptionalValue("selenium.remote", Boolean.class).orElse(false));
+
+        if (BooleanUtils.isTrue(deviceMode) && StringUtils.equalsIgnoreCase(Browser.CHROME.name(), browserType)) {
+            ChromeDeviceModeDriverProvider chromeDeviceModeDriverProvider = new ChromeDeviceModeDriverProvider();
+            Configuration.browser = chromeDeviceModeDriverProvider.getClass().getName();
+            Configuration.startMaximized = false;
```

Comment:
miert csak a chrome false?

Reply:
Eredeti helyről ahonnan szedve volt, ez volt. Kivezetem configba hogy állítható legyen.

Reply:
Ha devicemode true, akkor nem szabad fullscreenbe rakni, mert akkor elrontja a device mode-ot, és a képernyő maximális méretére rakja ki a browsert. Így nem lesz devicemode, tehát üti egymást a két setting, ezért volt false. Nem raknám ki.

Reply:
Amúgy nem Chrome, hanem Device, szóval mobil, tablet stb. azért van false-on most néztem meg jobban.

Reply:
Amikor ezt csináltam erre még nem volt native Selenide support. De időközben lett vmi setting selenide -be is. 
https://selenide.org/2019/11/29/selenide-5.5.1/
https://github.com/selenide/selenide/issues/1008

Reply:
valahogy ossze kellene kotni a megadott felbontas vs startMaximized opciot, Tehat ha megvan adva a felbontas akkor eleve false a maximized...

Reply:
Kicsit félrementek itt a dolgok. Szóval nem akarnék én alap felbontást megadni, az csak egy lehetőség amivel bemutattam hogy miként lehet tovább egyénre szabni a megnyitott böngészőt.

A Selenide-os megoldást pedig nem mondanám számunkra jónak, nem a legjobb System.getProperties -t használni és onnan olvasni fel a configot, azért vannak configurációs fájlok. Ezt preferáljuk itt.

SelenideConfig interfacebe kiszervezem az alap configokat, amit ott lehet majd bővíteni, ennek az implementálója majd Config-ból felszedi ezeket a részeket, és úgy adja vissza ahol kell, logokkal együtt.